src/cascadia/TerminalCore/Terminal.hpp

```diff
@@ -244,6 +246,15 @@ class Microsoft::Terminal::Core::Terminal final :
     std::function<void()> _pfnWarningBell;
     std::function<void(std::wstring_view)> _pfnTitleChanged;
     std::function<void(std::wstring_view)> _pfnCopyToClipboard;
+
+    // I've specifically put this instance here as it requires
+    //   alignas(std::hardware_destructive_interference_size)
+    // for best performance.
+    //
+    // But we can abuse the fact that the surrounding members rarely change and are huge
+    // (std::function is like 64 bytes) to create some natural padding without wasting space.
+    til::ticket_lock _readWriteLock;

```

Comment:
If you add the `alignas` now, would that make it safer (in case somebody changes the order) _and_ not compromise the layout by adding wasted padding space?

Reply:
I ask because... this is a micro-optimization somebody could easily accidentally remove or not fully understand and fail to maintain.

Reply:
`alignas` would add 56 bytes of padding space and force `_readWriteLock` to be 64 bytes large.
I had hoped that the comment I added there would be enough to prevent others from accidentally changing the order. It someone would move it to be next to the later members in this class it would double the overhead of the ticket lock (from 0.03% to 0.06% CPU usage), but I think the comment should be an okay-ish reminder to not do that. I hope?

Reply:
I think the comment is OK. I wonder if we'd ever be possessed to remove all the `std::functions` though and then the trick wouldn't work anymore.